pinot-common/src/main/java/org/apache/pinot/common/function/scalar/TrigonometricFunctions.java

```diff
@@ -0,0 +1,74 @@
+package org.apache.pinot.common.function.scalar;
+
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+
+public class TrigonometricFunctions {
```

Comment:
These can all be autovectorized on recent JVMs on x86, but need to be implemented as `TransformFunction`s. I would suggest generating the code with a one off script.